Sources/ComposableArchitecture/SwiftUI/Binding.swift

```diff
@@ -265,11 +265,18 @@ import SwiftUI
     public subscript<Value>(
       dynamicMember keyPath: WritableKeyPath<State, BindableState<Value>>
     ) -> Binding<Value>
-    where Action: BindableAction, Action.State == State, Value: Equatable {
-      self.binding(
-        get: { $0[keyPath: keyPath].wrappedValue },
-        send: { .binding(.set(keyPath, $0)) }
-      )
+      where Action: BindableAction, Action.State == State, Value: Equatable
+    {
+      ObservedObject(wrappedValue: self).projectedValue[bindable: keyPath]
```

Comment:
Could this trick be employed in `ViewStore.binding(get:send:)` instead? Then we could get it everywhere for free I think...

Reply:
Should be able to. I'll test it out.

Reply:
Hmmm... I think `subscript(bindable:)` is playing a large part here. Making it difficult with closures. Really want a simple solution for `ViewStore.binding(get:send:)`... back to the drawing board...

Reply:
@stephencelis thoughts on something like this?
```swift
final class WrappedState<LocalState>: ObservableObject {
  let get: () -> LocalState
  let set: (LocalState) -> Void

  init<State, Action>(
    viewStore: ViewStore<State, Action>,
    get toLocalState: @escaping (State) -> LocalState,
    send localStateToViewAction: @escaping (LocalState) -> Action
  ) {
    let initialState = toLocalState(viewStore.state)
    self.get = { [weak viewStore] in
      viewStore.flatMap { toLocalState($0.state) } ?? initialState
    }
    self.set = { [weak viewStore] newValue in
      viewStore?.send(localStateToViewAction(newValue))
    }
  }

  var state: LocalState {
    get { self.get() }
    set { self.set(newValue) }
  }

  var binding: Binding<LocalState> {
    ObservedObject(wrappedValue: self).projectedValue.state
  }
}
```
making `ViewStore.binding(get:send:)` 
```swift
public func binding<LocalState>(
  get: @escaping (State) -> LocalState,
  send localStateToViewAction: @escaping (LocalState) -> Action
) -> Binding<LocalState> {
  WrappedState(viewStore: self, get: get, send: localStateToViewAction).binding
}
```

Reply:
@iampatbrown Looks cool! The only thing that pops out is ownership, like when is `viewStore` unavailable to this observable object and when would it fall back to initial state. Did you encounter a situation that ran that code path?

Reply:
I don't think I'd end up in a situation where the binding would exist without the `viewStore`. I was just being over cautious. Would you recommend using a strong reference here?

Reply:
I don't overly love the idea of wrapper class like this... so far it's all I've got that has the expected binding behaviour

Reply:
Ha I have the same knee-jerk reaction, but it's a lean class, and it's the best way so far to reproduce the behavior (thanks!). Ideally we benchmark it, but I doubt it adds much overhead.

I think the ideal would be for SwiftUI to document and offer APIs to derive custom bindings, but in lieu of that, reliable/predictable workarounds are definitely valuable.

Reply:
Cool. I'd be interested to learn how you'd go about bench marking something like this.

So far I've just isolated the implementation and compared some basic tasks using `Time Profiler` and `swift-benchmark`. I repeated the process with the TCA implementations as well. 

The current `Binding.init` seems to perform the best, but that's mainly because it isn't accessing the `ObservableObjectLocation`. The `WrappedState` method preforms pretty similar to using the native `ObservedObject` bindings. In fact, `store.binding(get: { $0.text }, send: { .textChanged($0) })` appeared better than `observedStore.projectedValue.state.text`. Probably to do with keyPath performance vs closures. 

Anyway, I'll leave this one for now. If you need me to do anything, just let me know.